internal/batches/service/service.go

```diff
@@ -52,11 +52,19 @@ func New(opts *Opts) *Service {
 	}
 }
 
+// The reason we ask for batchChanges here is to surface errors about trying to use batch
+// changes in an unsupported environment sooner, since the version check is typically the
+// first thing we do.
 const sourcegraphVersionQuery = `query SourcegraphVersion {
 	site {
 	  productVersion
 	}
+	batchChanges(first: 1) {
+    nodes {
+      id
+    }
   }
+}
```

Comment:
tabs vs spaces war going on here :)

Reply:
I wonder if this is a bit of a dirty hack? I don't have strong feelings about it though. Might be nicer to expose a property "batchChangesEnabled" or something like that that would on dotcom always be falsy and on instances where BC is turned off via site config as well. But it wouldn't work with older SG instances .. so .. well let's not overengineer this 🤷‍♂️

Reply:
Yeah, I might prefer a new field that we can query for this just so we don't have to hit the database, but I don't think that's critical here, particularly since we'd have to "feature" it to depend on a particular version. Could be a nice "first issue" type sustaining ticket for someone else.

Reply:
Yeah I definitely agree this is a better idea but felt a bit overkill for the time being. 😅 I think there are other ways we could improve the UX if you try to run `src batch preview|apply` on dotcom/an unsupported environment beyond just failing earlier, so I'll log this under an umbrella issue for those other ideas I also had. 🙂

Reply:
https://github.com/sourcegraph/src-cli/issues/690